packages/paima-sdk/concise/src/delegate.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  BuiltinGrammar,
+  BuiltinGrammarPrefix,
+  KeyedBuiltinGrammar,
+  parseStmInput,
+} from "@paima/concise";
+import { privateKeyToAccount } from "viem/accounts";
+import { createWalletClient, http } from "viem";
+
+export function extractDelegateWallet(inputData: string) {
+  const parsed = parseStmInput<
+    typeof BuiltinGrammar,
+    | typeof BuiltinGrammarPrefix.createAccount
+    | typeof BuiltinGrammarPrefix.linkAddress
+    | typeof BuiltinGrammarPrefix.unlinkAddress
+  >(
+    inputData,
+    BuiltinGrammar,
+    KeyedBuiltinGrammar,
+  );
+  return parsed;
+}
+
+export const accountMessages = {
+  linkAccount: (
+    account_id: number,
+    other_address: string,
+    is_new_primary: boolean,
+  ) => {
+    return `link:${String(account_id)}:${other_address}:${
+      is_new_primary ? "true" : "false"
+    }`;
+  },
+  unlinkAccountWithPrimary: (
+    account_id: number,
+    other_address: string,
+    new_primary?: string | null,
+  ) => {
+    return `unlink:${String(account_id)}:${other_address}:${new_primary ?? ""}`;
+  },
+};
+
+export const signMessage = async (
+  message: string,
+  privateKey: `0x${string}`,
+) => {
+  const account = privateKeyToAccount(privateKey);
+  const walletClient = createWalletClient({
+    account,
+    transport: http("http://0.0.0.0"),
```

Comment:
TODO

Reply:
viem requires with this method a http url or gives a warning in the terminal, this is also not required if chain is defined as it has a defined url